ROOTFS/etc/waggle/sanity/fatal/core_sdcard.test

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+pf="Recovery SDCard test"
+echo "$pf: Begin"
+
+# run fsck on our sd card
+fsck -n -f /dev/mmcblk1p1
+
+if [ "$?" != "0" ];
+then
+        echo "$pf: /dev/mmcblk1p1 failed fsck FAIL"
+        exit 1
+fi
+
+fsck -n -f /dev/mmcblk1p2
+
+if [ "$?" != "0" ];
+then
+        echo "$pf: /dev/mmcblk1p2 failed fsck FAIL"
+        exit 2
+fi
+
+
+mkdir /tmp/core_sdcard_test
+
+mount -r /dev/mmcblk1p1  /tmp/core_sdcard_test/   
+
+if [ "$(cat /tmp/core_sdcard_test/etc/overlayroot.conf | grep "overlayroot=\"tmpfs")" = ""  ];
+then
+        echo "$pf: sdcard tmpfs not properly set up FAIL"
+        exit 3
+fi
+
+chroot /tmp/core_sdcard_test/ ls
```

Comment:
if you are going to do it this way

```
chroot /tmp/core_sdcard_test/ ls
[ "$?" != "0" ];
```

then these commands should be right next to each other to show they are affiliated.

OR you could do this more cleanly with

```
if ! chroot /tmp/core_sdcard_test/ ls; then
        echo "$pf: Could not run chroot on sdcard, invalid rootfs FAIL"
        exit 4
fi
```